internal/resources/collection.go

```diff
@@ -0,0 +1,394 @@
+	providerschema "github.com/couchbasecloud/terraform-provider-couchbase-capella/internal/schema"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &Collection{}
+	_ resource.ResourceWithConfigure   = &Collection{}
+	_ resource.ResourceWithImportState = &Collection{}
+)
+
+const errorMessageWhileCollectionCreation = "There is an error during collection creation. Please check in Capella to see if any hanging resources" +
+	" have been created, unexpected error: "
+
+const errorMessageAfterCollectionCreation = "Collection creation is successful, but encountered an error while checking the current" +
+	" state of the collection. Please run `terraform plan` after 1-2 minutes to know the" +
+	" current collection state. Additionally, run `terraform apply --refresh-only` to update" +
+	" the state from remote, unexpected error: "
+
+// Collection is the collection resource implementation.
+type Collection struct {
+	*providerschema.Data
+}
+
+// NewCollection is a helper function to simplify the provider implementation.
+func NewCollection() resource.Resource {
+	return &Collection{}
+}
+
+// ImportState imports a remote collection that is not created by Terraform.
+func (c *Collection) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	// Retrieve import name and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("collection_name"), req, resp)
+}
+
+// Metadata returns the Collection resource type name.
+func (c *Collection) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_collection"
+}
+
+// Schema defines the schema for the Collection resource.
+func (c *Collection) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = CollectionSchema()
+}
+
+// Configure It adds the provider configured api to the collection resource.
+func (c *Collection) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	data, ok := req.ProviderData.(*providerschema.Data)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *ProviderSourceData, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	c.Data = data
+}
+
+// Create creates a new collection.
+func (c *Collection) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.Collection
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	collectionRequest := collection_api.CreateCollectionRequest{
+		Name:   plan.Name.ValueString(),
+		MaxTTL: plan.MaxTTL.ValueInt64(),
```

Comment:
If MaxTTL is an optional value, then where are we checking whether MaxTTL is populated in the plan or not?

Reply:
Done